test/unit/spec/narrative_core/jobCommChannel-spec.js

```diff
@@ -380,86 +339,106 @@ define(['jobCommChannel', 'base/js/namespace', 'common/runtime'], (
+                        }
+                    );
                 });
             });
         });
 
-        it('Handle unknown job errors generically', (done) => {
+        it('Handle unknown job errors generically', () => {
             const jobId = 'jobWithErrors',
                 errMsg = 'some random error',
                 requestType = 'some-error',
-                msg = makeCommMsg('job_comm_error', {
+                busMsg = {
                     source: requestType,
                     job_id: jobId,
                     message: errMsg,
-                }),
+                },
+                msg = makeCommMsg('job_comm_error', busMsg),
                 comm = new JobCommChannel();
-            channelChecker('jobId', jobId, 'job-error', done, {
-                jobId: jobId,
-                message: errMsg,
-                request: requestType,
-            });
-            comm.initCommChannel().then(() => {
+            spyOn(this.bus, 'send');
+            return comm.initCommChannel().then(() => {
                 comm.handleCommMessages(msg);
+                expect(this.bus.send).toHaveBeenCalledWith(
+                    {
+                        jobId: jobId,
+                        message: errMsg,
+                        request: requestType,
+                    },
+                    {
+                        channel: { jobId: jobId },
+                        key: { type: 'job-error' },
+                    }
+                );
             });
         });
 
-        it('Should handle job_init_err and job_init_lookup_err', (done) => {
-            let count = 0;
-            ['job_init_err', 'job_init_lookup_err'].forEach((errType) => {
+        ['job_init_err', 'job_init_lookup_err'].forEach((errType) => {
+            it(`Should handle ${errType}`, () => {
                 const msg = makeCommMsg(errType, {
                         service: 'job service',
                         error: 'An error happened!',
                         name: 'Error',
                         source: 'jobmanager',
                     }),
                     comm = new JobCommChannel();
-                comm.initCommChannel()
+                jasmine.clock().install();
+                return comm
+                    .initCommChannel()
                     .then(() => {
                         comm.handleCommMessages(msg);
-                        return new Promise((resolve) => {
-                            setInterval(resolve, 1000);
-                        });
                     })
                     .then(() => {
-                        expect(document.querySelector('#kb-job-err-report')).not.toBeNull();
-                        count++;
-                        if (count == 2) {
-                            done();
-                        }
+                        jasmine.clock().tick(5000);
+                        expect(document.querySelector('.modal #kb-job-err-trace')).not.toBeNull();
```

Comment:
is there a built-in time delay for these modals popping up and being dismissed? Or are you just waiting for the DOM to finish its transactions here?

Reply:
There's a fade in and out and the event is tied to the modal object (BootstrapModal class), which isn't exposed. Forcing the clock to tick helps with the wait to remove it.

Reply:
...and that doesn't actually work, it really just works on things like `setTimeout`. I just got a failure where that modal wasn't removed.

Hm. Figuring something else out is tricky while maintaining some unit testing and not digging into implementation details.

Reply:
We will have to come up with a pattern for dealing with potentially delaying DOM alterations -- I'm not sure if there are many solutions beyond using `setTimeout(() => { expect(...) }, delay)` in the tests. Promisify all the DOM changes? 😉 No, that way madness lies!

Reply:
I don't like it, but doing the triggers, then returning a promise that only resolves after a `setTimeout` fixes it.  :\
I've pushed the change.